File: kotlin-java/src/main/java/me/zeroest/kotlin/basic/classproperty/PropertyStatus.java
package me.zeroest.kotlin.basic.classproperty;

public class PropertyStatus {

    private static class JavaCoffee {
        private String name = "Java Coffee";
        private boolean iced;

        public String getName() {
            return name;
        }

        public boolean isIced() {
            return iced;
        }

        public void setIced(boolean iced) {
            this.iced = iced;
        }
    }

    public static void main(String[] args) {
        JavaCoffee javaCoffee = new JavaCoffee();

        javaCoffee.setIced(true);

        if (javaCoffee.isIced()) { // 상태를 메서드로 표현
            System.out.printf("Ice %s\n", javaCoffee.getName());
        } else {
            System.out.printf("Hot %s\n", javaCoffee.getName());
        }
    }

}

File: kotlin-java/src/main/kotlin/me/zeroest/kotlin/basic/classproperty/PropertyStatus.kt
package me.zeroest.kotlin.basic.classproperty

class ProCoffee(
    var name: String = "Pro Coffee",
    var price: Int = 0,
    var iced: Boolean = false,
)

fun main() {
    val proCoffee = ProCoffee()

    // 프로퍼티를 사용해 상태를 나타낼 수 있기 때문에 자바보다 객체지향적으로 코드를 작성할 수 있다
    proCoffee.iced = true
    if (proCoffee.iced) { // 객체의 상태를 프로퍼티로 표현
        println("Ice ${proCoffee.name}")
    } else {
        println("Hot ${proCoffee.name}")
    }
}
